src/Versatile_RotaryEncoder.cpp

```diff
@@ -147,9 +151,14 @@ bool Versatile_RotaryEncoder::ReadEncoder() {
             switch (button) {
                 case switchup:
                     encoder = release;
-                    if (handlePressRelease != nullptr) {
-                        handlePressRelease();
-                        handled_functions = true;
+                    if(check_double_press && (millis() - last_switchup < double_press_duration)) {
+                        check_double_press = false;
+                        if (handleDoublePressRelease != nullptr) {
+                            handleDoublePressRelease();
+                            handled_functions = true;
+                        }
+                    } else {
+                        check_double_press = true;
```

Comment:
"check_double_press" needs to implemented not as a flag but as "enum Button" instead to be used on the present switch case.